api/envoy/service/ext_proc/v3/external_processor.proto

```diff
@@ -119,6 +110,19 @@ message ProcessingRequest {
   // in the list is populated from the standard
   // :ref:`attributes <arch_overview_attributes>` supported across Envoy.
   map<string, google.protobuf.Struct> attributes = 9;
+
+  // Specify whether the filter that sent this request is running in :ref:`observability_mode
+  // <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.observability_mode>`
+  // and defaults to false.
+  //
+  // * A value of ``false`` indicates that the server must respond
+  //   to this message by either sending back a matching ProcessingResponse message,
+  //   or by closing the stream.
+  // * A value of ``true`` indicates that the server must not respond to this
+  //   message, although it may still close the stream to indicate that no more messages
+  //   are needed.
```

Comment:
what does it mean when you say the server must not respond? are we going to fire an error when server responds ? or are we simply ignoring the response (i.e. not even reading from that socket, in which case, server will have send() errors) ?

Reply:
Basically, this establishes the contract that server don't respond to such request message.

If they break the contract, ext_proc filter still receive the response (i.e., read from the socket) but simply ignore it without applying any actions (e.g., mutation etc).  

Regarding the error handling, 
(1) Returning error to ext_proc server probably requires sending another request . It is different from the streamError that we can do for oversized response as that is more integrated in lower gRPC layer and here is more about application layer logic.
(2) Another way of handling it : file error and sendLocalReply to downstream client. ext_proc currently use this approach for [out-of-order/spurious response](https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/ext_proc/ext_proc.cc#L944-L955)

(1) probably is best for observability but we need to balance the increased complexity to the protocol, which probably fall into discussion on the potential observability improvement along with other similar cases (e.g. out-of-order response etc). 
(2) is probably too strict for observability mode as that basically reject the client request. 

So I am currently leaning towards `ignore the response and continue` but definitely open to other ideas.

Reply:
why do we even have to read from the socket if ext_proc filter is in observability mode? Put another way, does reading result in any perf impact? If it does not, then we are good
wrt how to handle servers that send responses when in observability_mode, I think option 3 is good.

Reply:
I meant to say : ext_proc filter still receive message but doesn't read that message. i.e., Ignore the response on `onReceiveMessage`

Maybe `read the message` is what you mean by `reading from the socket`? I may misunderstood it as sockets in lower transport layer.

Reply:
Connection to the ext_proc can be shared by multiple ext_proc request, some of which are in the observability mode and some are not. Envoy still needs to read from the ext_proc server. Stating the responses are ignored should be fine.